conformance/scenarios/httpgateway/httpgateway.yaml

```diff
@@ -0,0 +1,45 @@
+kind: Gateway
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: my-gateway
```

Comment:
namespace may be important here. For in-cluster implementations this is presumably going to require a deployment/service in the same namespace. Unless their is some operator provisioning these in response to Gateways, then we need some way to provision these prior to running conformance test, which means we need to know the namespace, which I think is random?

Reply:
Yeah I've been thinking about that. For prior art here, ingress-controller-conformance automatically generated these resources based on the references in Ingress. I think we'll likely want to just embed the Service and Deployment references in the YAML manifests so we can also test for missing/invalid refs. (It may be simpler to just create a different test path/func for that purpose, not quite sure yet).

As far as namespaces, we'll also need some more advanced capabilities here when compared with ingress-controller-conformance. We'll need to be able to generate multiple namespaces and have a way to describe which resources belong in each of those namespaces. Not quite sure what the cleanest way to do that is, but it's something that there's no prior art for due to the mostly single namespace capabilities of ingress.